src/App.tsx

```diff
@@ -1,35 +1,75 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import './App.css';
+import CarForm from './components/car/car_form';
+import { useState } from 'react';
+import CarList from './components/car/car_list';
+import { Car } from './types';
+import MaterialUISwitch from './components/materialUISwitch'; 
+import { createTheme, ThemeProvider } from '@mui/material';
+import {  useEffect } from 'react';
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [cars, setCars] = useState<Car[]>([]);
+  const [darkMode, setDarkMode] = useState<boolean>(false);
+
+  const addCar = (car: Car) => {
+    setCars([...cars, car]);
+  };
+
+  const toggleDarkMode = () => {
+    setDarkMode(!darkMode);
+  };
+
+  const theme = createTheme({
+    palette: {
+      mode: darkMode ? 'dark' : 'light',
+      primary: {
+        main: '#646bf3',
+      },
+      secondary: {
+        main: '#fff',
+      },
+      background: {
+        default: darkMode ? '#101113' : '#fff',
+      },
+      text: {
+        primary: darkMode ? '#fefefe' : '#101113',
+      },
+    },
+  });
+
+  useEffect(() => {
+    document.body.style.backgroundColor = theme.palette.background.default;
+    document.body.style.color = theme.palette.text.primary;
+  }, [theme]);
```

Comment:
Instead of changing the color of body, could you try to wrap your app into some  MUI component that will be using background color from palette? It may be <Paper>, <Box> or anything else MUI provide